openapi_tester/clients.py

```diff
@@ -33,6 +34,36 @@ def request(self, **kwargs) -> Response:  # type: ignore[override]
         self.schema_tester.validate_response(response)
         return response
 
+    # pylint: disable=W0622
+    def post(self, path, data=None, format=None, content_type="application/json", follow=False, **extra):
```

Comment:
Adding another argument would still require to keep the other methods I think, but we can check if the object is serializable and then based on that serialize it or not. This would maintain the backwards compatibility, while also getting rid of the other methods e.g.:

```
    @staticmethod
    def _serialize(data):
        try:
            return json.dumps(data)
        except (TypeError, OverflowError):
            """Data is already serialized"""
            return data

    def request(self, **kwargs) -> Response:  # type: ignore[override]
        """Validate fetched response against given OpenAPI schema."""
        if "application/json" in kwargs["content_type"]:
            kwargs["data"] = self._serialize(kwargs["json"])
        response = super().request(**kwargs)
        if self._is_successful_response(response):
            self.schema_tester.validate_request(response)
        self.schema_tester.validate_response(response)
        return response
```

Let me know what do you think

Reply:
Ah, it actually wouldn't work, because the data in kwargs is different than the initial data. This is because it's actually an input stream and it can only be read once (from what I understand), so we can't serialize it there. However, I can add that _serialize() function to the existing methods, which would still resolve the breaking change issues

Reply:
You are correct, `data` is indeed altered already and not the same that is passed to `request` method (https://github.com/django/django/blob/main/django/test/client.py#L663)

I would say yes using `_serialize` could work along with also checking if the `header` is passed would be good for avoiding the breaking change.

Reply:
Yes sorry, I wanted to mean indeed the `content-type` check, which you had already included. Change is great as it is 👍